plane/src/types/mod.rs

```diff
@@ -70,6 +70,53 @@ impl FromStr for ClusterName {
     }
 }
 
+#[derive(
+    Default, Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash, valuable::Valuable,
+)]
+pub struct DronePoolName(String);
+
+impl DronePoolName {
+    pub fn is_default(&self) -> bool {
+        self == &Self::default()
+    }
+
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+}
+
+impl From<String> for DronePoolName {
+    fn from(s: String) -> Self {
+        DronePoolName(s)
+    }
+}
+
+impl From<&str> for DronePoolName {
+    fn from(s: &str) -> Self {
+        DronePoolName(s.to_string())
+    }
+}
+
+impl From<&Option<String>> for DronePoolName {
```

Comment:
This is only used in the platform. Does it make sense to keep here or only implement in the platform?

Reply:
A trait `impl` has to be in the same module as either the `trait` or the type being implemented for (`DronePoolName` here), to avoid ambiguous impls. That said, it might be more explicit just to implement `From<String>`, and the call site call `optional_pool.map(DronePoolName::from).unwrap_or_default()`.

(`From<&str>` would also work, I have a slight preference for `From<String>` because `From<String>` will compile to a no-op whereas `From<&str>` will allocate and copy the string even if the original is never used again.)